src/sc/sidechain.cpp

```diff
@@ -757,61 +778,139 @@ void CSidechain::UpdateScFees(const CScCertificateView& certView)
     LogPrint("sc", "%s():%d - pushing f=%d/m=%d into list with size %d\n",
         __func__, __LINE__, ftScFee, mbtrScFee, scFees.size());
 
-    scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
+    // v1 sidechains
+    if (!isNonCeasing()) {
+        scFees.push_back(Sidechain::ScFeeData(ftScFee, mbtrScFee));
 
-    // check if we are past the buffer size
-    int delta = scFees.size() - maxSizeOfScFeesContainers;
-    if (delta > 0)
-    {
-        // remove from the front as many elements are needed to be within the circular buffer size
-        // --
-        // usually this is just one element, but in regtest a node can set the max size via a startup option
-        // therefore such size might be lesser than scFee size after a node restart
-        while (delta > 0)
+        int delta = scFees.size() - maxSizeOfScFeesContainers;
+        if (delta > 0)
```

Comment:
The `if (delta > 0)` is not needed here imho. Having no `else` branch the `while (delta > 0)` is enough

Reply:
Fixed in f0bab99cd7559a9bab4c8dc65103df6eea2b4b76